assets/js/state/index.js

```diff
@@ -45,104 +45,50 @@ const joinChannel = (channel) => {
     .receive('timeout', () => logMessage('Networking issue. Still waiting...'));
 };
 
-const processChannelEvents = (store) => {
-  const socket = new Socket('/socket', {});
-  socket.connect();
-
-  const hostsChannel = socket.channel('monitoring:hosts', {});
-  const clustersChannel = socket.channel('monitoring:clusters', {});
-  const sapSystemsChannel = socket.channel('monitoring:sap_systems', {});
-  const databasesChannel = socket.channel('monitoring:databases', {});
-
-  hostsChannel.on('host_registered', (payload) =>
-    store.dispatch({ type: 'HOST_REGISTERED', payload })
-  );
-
-  hostsChannel.on('host_details_updated', (payload) =>
-    store.dispatch({ type: 'HOST_DETAILS_UPDATED', payload })
-  );
-
-  hostsChannel.on('heartbeat_succeded', (payload) =>
-    store.dispatch({ type: 'HEARTBEAT_SUCCEDED', payload })
-  );
-
-  hostsChannel.on('heartbeat_failed', (payload) =>
-    store.dispatch({ type: 'HEARTBEAT_FAILED', payload })
-  );
-
-  clustersChannel.on('cluster_registered', (payload) =>
-    store.dispatch({ type: 'CLUSTER_REGISTERED', payload })
-  );
-
-  clustersChannel.on('cluster_details_updated', (payload) =>
-    store.dispatch({ type: 'CLUSTER_DETAILS_UPDATED', payload })
-  );
-
-  clustersChannel.on('checks_execution_started', (payload) =>
-    store.dispatch({ type: 'CHECKS_EXECUTION_STARTED', payload })
-  );
-
-  clustersChannel.on('checks_execution_completed', (payload) =>
-    store.dispatch({ type: 'CHECKS_EXECUTION_COMPLETED', payload })
-  );
-
-  clustersChannel.on('checks_results_updated', (payload) =>
-    store.dispatch({ type: 'CHECKS_RESULTS_UPDATED', payload })
-  );
+const registerEvents = (socket, channelName, events) => {
+  const channel = socket.channel(channelName, {});
 
-  clustersChannel.on('cluster_health_changed', (payload) =>
-    store.dispatch({ type: 'CLUSTER_HEALTH_CHANGED', payload })
-  );
+  for (const event of events) {
+    channel.on(event, (payload) =>
+      store.dispatch({ type: event.toUpperCase(), payload })
```

Comment:
We don't have local access to the `store` variable here. You could pass it as a parameter to the `processChannelEvents` function and to the `registerEvents` function